src/hub/dataSources/schema/PhotonSchema.ts

```diff
@@ -0,0 +1,138 @@
+import Log from "../../../shared/log/Log";
+
+export default function process(log: Log, key: string, timestamp: number, value: Uint8Array) {
+  let result = parsePacket(value, timestamp);
+  saveResult(log, key, timestamp, result);
+}
+
+/** Parses raw data to create a pipeline result. */
+function parsePacket(value: Uint8Array, timestamp: number): PhotonPipelineResult {
+  let view = new DataView(value.buffer, value.byteOffset, value.byteLength);
+  let offset = 0;
+
+  let result = new PhotonPipelineResult();
+
+  result.latency = view.getFloat64(offset);
+  result.timestamp = timestamp - result.latency;
+  offset += 8;
+
+  const numTargets = view.getInt8(offset);
+  offset += 1;
+
+  result.targets = [];
+  for (let i = 0; i < numTargets; i++) {
+    let target = new PhotonTrackedTarget();
+    target.yaw = view.getFloat64(offset);
+    offset += 8;
+    target.pitch = view.getFloat64(offset);
+    offset += 8;
+    target.area = view.getFloat64(offset);
+    offset += 8;
+    target.skew = view.getFloat64(offset);
+    offset += 8;
+    target.fiducialId = view.getInt32(offset);
+    offset += 4;
+
+    target.bestCameraToTarget = parseTransform3d(view, offset);
+    offset += 7 * 8;
+    target.altCameraToTarget = parseTransform3d(view, offset);
+    offset += 7 * 8;
+    target.poseAmbiguity = view.getFloat64(offset);
+    offset += 8;
+
+    target.minAreaRectCorners = [];
+    for (let j = 0; j < 4; j++) {
+      let x = view.getFloat64(offset);
+      offset += 8;
+      let y = view.getFloat64(offset);
+      offset += 8;
+      target.minAreaRectCorners.push({ x: x, y: y });
+    }
+
+    target.detectedCorners = [];
+    const numCorners = view.getInt8(offset);
+    offset += 1;
+    for (let j = 0; j < numCorners; j++) {
+      let x = view.getFloat64(offset);
+      offset += 8;
+      let y = view.getFloat64(offset);
+      offset += 8;
+      target.detectedCorners.push({ x: x, y: y });
+    }
+
+    result.targets.push(target);
+  }
+
+  return result;
+}
+
+/** Saves a pipeline result to a log file. */
+function saveResult(log: Log, key: string, timestamp: number, result: PhotonPipelineResult) {
+  log.putNumber(key + "/latency", timestamp, result.latency);
+  log.putNumber(key + "/timestamp", timestamp, result.timestamp);
+
+  for (const [idx, target] of result.targets.entries()) {
```

Comment:
LOL this curst code -- is there a more TS way to do this?

Reply:
Well, you could do `result.targets.forEach(target => {})` and then explicitly have a log call for each field. For the target x/y a filter might be appropriate, like `value.detectedCorners.filter((element, index) => index % 2 == 0)`